server/src/main/scala/org/labrad/browser/RegistryApi.scala

```diff
@@ -22,7 +22,19 @@ object RegistryApi {
   }
 }
 
+/**
+ * The contents of a registry directory, potentially including recursive subdirectory contents
+ */
+case class RegistryContents(dirs: Seq[(String, RegistryContents)], keys: Seq[(String, Data)])
+object RegistryContents {
+  val Empty = RegistryContents(dirs = Nil, keys = Nil)
+}
```

Comment:
An object with the same name as a trait or class is called a "companion object" in scala. This is a useful place to put things that are associated with the class as a whole, rather than associated with a specific instance of the class. In java, one would use static members for this instead, but scala doesn't have statics.

For case classes, the compiler automatically generates a companion object with various convenience features, like an apply method for creating instances (so you can say `MyCaseClass(x)` to construct an instance, which actually calls a method on the companion object `MyCaseClass.apply(x)`, instead of calling the constructor like `new MyCaseClass(x)`).

In the code here, I've explicitly defined a companion object to hold a special empty instance of `RegistryContents`. But in addition, the compiler adds the autogenerated members as usual for case class companion objects; for example you can see that I'm using the companion object apply method to create the Empty instance.